### PR TITLE
Fix #3751. Set to use default db path from definition.

### LINKED ISSUE #3751: We have keywords.db and metadata.db, we need to remove one
### Problem

We have metadata.db : https://github.com/inasafe/inasafe/blob/develop/safe/metadata/metadata_db_io.py#L61

And keyword.db : 
https://github.com/inasafe/inasafe/blob/develop/safe/utilities/keyword_io.py#L249

### Expected Behaviour
We should remove one I think. @ismailsunni Can you confirm ?

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #3751 
* Funded by: DFAT
* Description: 
  - Use default db path from definition
  - Clean KeywordIO

### CODE REVIEW

**Comment by Gustry** [comment]
Can it go in 4.1.1? I will check your pr later. I'm in the airport. Nearly in paris

**Comment by ismailsunni** [comment]
@gustry yes, it's fine to go 4.1.1. Thanks

File: safe/definitions/default_settings.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
     'ISO19115_LICENSE': 'Free use with accreditation',
 
     'keywordCachePath': abspath(join(
-        expanduser('~'), '.inasafe', 'keywords.db')),
+        expanduser('~'), 'inasafe', 'metadata.db')),
```
Comment (Gustry) [inline safe/definitions/default_settings.py]: Why do you remove the hidden folder ?
Anyway, we shouldn't use a folder in the user directory. We should use the QGIS user directory

Comment (ismailsunni) [inline safe/definitions/default_settings.py]: Ah yes, thanks. I copy and paste from the metadata. But it seems I pasted the wrong one.

Comment (Gustry) [inline safe/definitions/default_settings.py]: Use `QgsApplication.qgisSettingsDirPath()`

**Comment by Gustry** [comment]
LGTM thanks
it fixes https://github.com/inasafe/inasafe/issues/3214 too I think

**Pull request merged**